DynamicSmoothedGUI.py

```diff
@@ -210,14 +215,88 @@ def checkRecalc(self):
+
+    def get_direction_coor(self, curr_x, curr_y, angle):
+
+        x2 = math.cos(math.radians(angle+90))* vis_radius + curr_x
+        y2 = math.sin(math.radians(angle+90)) * vis_radius + curr_y
+        return (x2 / tile_scale_fac, y2 / tile_scale_fac)
+
+    def draw_line(self, curr_x, curr_y, next_x, next_y):
+        self.canvas.create_line(curr_x / tile_scale_fac, curr_y / tile_scale_fac, next_x / tile_scale_fac,
+                                next_y / tile_scale_fac, fill="#339933", width=1.5)
+
     def updateGridSmoothed(self):
         """
         updates the grid in a smoothed fashion
         """
         try:
-            # If this is the first tile loop is being iterated through we need to initialize
-            if self.initPhase:
-                print('IN INIT PHASE')
+            if self.desired_heading is not None and self.heading == self.desired_heading:
+                self.canvas.delete(self.angle_trace)
+                line_coor = self.get_direction_coor(self.curr_tile.x, self.curr_tile.y, self.heading)
+                self.angle_trace = self.canvas.create_line(self.curr_tile.x / tile_scale_fac, self.curr_tile.y / tile_scale_fac, line_coor[0], line_coor[1], fill='#FF69B4', width=1.5)
+                self.canvas.delete(self.des_angle_trace)
+                des_line_coor = self.get_direction_coor(self.curr_tile.x, self.curr_tile.y, self.desired_heading)
+                self.des_angle_trace = self.canvas.create_line(self.curr_tile.x / tile_scale_fac, self.curr_y / tile_scale_fac, des_line_coor[0],
+                                                               des_line_coor[1], fill = '#FF0000', width=1.5)
+                self.canvas.pack()
+                self.output_state = "move forward"
+                print(self.output_state)
+
+            if self.desired_heading is not None and self.heading != self.desired_heading:
+                self.canvas.delete(self.angle_trace)
+                self.canvas.delete(self.des_angle_trace)
+                line_coor = self.get_direction_coor(self.curr_tile.x, self.curr_tile.y, self.heading)
+                print("drawing line from " + str(self.curr_tile.x / tile_scale_fac) + ', ' + str(self.curr_tile.y / tile_scale_fac) + ' to: ' + str(line_coor[0] / tile_scale_fac) + str(line_coor[1] / tile_scale_fac))
+                self.angle_trace = self.canvas.create_line(self.curr_tile.x / tile_scale_fac, self.curr_tile.y / tile_scale_fac, line_coor[0],
+                                                           line_coor[1], fill='#FF69B4', width=1.5)
+                des_line_coor = self.get_direction_coor(self.curr_tile.x, self.curr_tile.y, self.desired_heading)
+                self.des_angle_trace = self.canvas.create_line(self.curr_tile.x / tile_scale_fac, self.curr_y / tile_scale_fac, des_line_coor[0],
+                                                               des_line_coor[1], fill = '#FF0000', width=1.5)
+                self.canvas.pack()
```

Comment:
looks like repeated code from before; consider making a helper function to call for both cases instead

Reply:
Done